pkg/querier/querier.go

```diff
@@ -88,25 +90,32 @@ func New(cfg Config, distributor Distributor, chunkStore ChunkStore) (storage.Qu
 }
 
 // NewQueryable creates a new Queryable for cortex.
-func NewQueryable(dq, cq storage.Queryable, distributor Distributor) storage.Queryable {
+func NewQueryable(dq, cq storage.Queryable, distributor Distributor, ingesterMaxChunkAge time.Duration) storage.Queryable {
 	return storage.QueryableFunc(func(ctx context.Context, mint, maxt int64) (storage.Querier, error) {
-		dqr, err := dq.Querier(ctx, mint, maxt)
-		if err != nil {
-			return nil, err
+		q := querier{
+			distributor: distributor,
+			ctx:         ctx,
+			mint:        mint,
+			maxt:        maxt,
 		}
 
 		cqr, err := cq.Querier(ctx, mint, maxt)
 		if err != nil {
 			return nil, err
 		}
 
-		return querier{
-			queriers:    []storage.Querier{dqr, cqr},
-			distributor: distributor,
-			ctx:         ctx,
-			mint:        mint,
-			maxt:        maxt,
-		}, nil
+		// Include ingester only if maxt is within 2 times ingester chunk age w.r.t. current time.
+		if maxt >= time.Now().Add(-2*ingesterMaxChunkAge).UnixNano()/1e6 {
+			dqr, err := dq.Querier(ctx, mint, maxt)
+			if err != nil {
+				return nil, err
+			}
+			q.queriers = []storage.Querier{dqr, cqr}
+		} else {
+			q.queriers = []storage.Querier{cqr}
+		}
```

Comment:
Super nit, I would remove the else and use `append` instead.

Reply:
That is exactly what I had before changing at the end! Will change it to append.